cpp/src/reductions/scan/scan_inclusive.cu

```diff
@@ -156,17 +158,10 @@ struct scan_functor<Op, cudf::string_view> {
 template <typename Op>
 struct scan_dispatcher {
  private:
-  template <typename T>
-  static constexpr bool is_min_max_supported()
-  {
-    return cudf::is_relationally_comparable<T, T>() && !cudf::is_dictionary<T>() &&
-           (std::is_same_v<Op, DeviceMin> || std::is_same_v<Op, DeviceMax>);
-  }
-
   template <typename T>
   static constexpr bool is_supported()
   {
-    return std::is_arithmetic_v<T> || is_min_max_supported<T>();
+    return std::is_invocable_v<Op, T, T> && !cudf::is_dictionary<T>();
```

Comment:
It really makes me happy that all that was needed here was to just change how `is_supported` was implemented and suddenly whole new type/op combinations are automatically enabled without any other work.

Reply:
I'm curious, if `T` is dictionary then will `std::is_invocable_v<Op, T, T>` returns `true`? If not, we also don't need to check `!cudf::is_dictionary<T>()` anymore.

Reply:
Yes, `std::is_invocable` will be true for dictionary types on certain operations but dictionary is purposely not supported for scan.